python/iot-msk-lambda-pipeline/msk_demo/msk_demo_stack.py

```diff
@@ -0,0 +1,324 @@
+            edition=ec2.AmazonLinuxEdition.STANDARD,
+            virtualization=ec2.AmazonLinuxVirt.HVM,
+            storage=ec2.AmazonLinuxStorage.GENERAL_PURPOSE
+        )
+        
+        role = iam.Role(self, "InstanceSSM", assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"))
+
+        role.add_managed_policy(iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSSMManagedInstanceCore"))
+
+        # Instance
+        instance = ec2.Instance(self, "Instance",
+            instance_type=ec2.InstanceType(constants["KAFKA_CLIENT_INSTANCE"]),
+            machine_image=amzn_linux,
+            vpc = vpc,
+            vpc_subnets=client_subnet,
+            role = role,
+            
+        )
+
+        client_security_group = ec2.SecurityGroup(self, 'InstanceSecurityGroup', vpc=vpc)
+        
+        client_security_group.add_ingress_rule(
+            ec2.Peer.ipv4('0.0.0.0/0'),
+            ec2.Port.tcp(22),
+        )
+        instance.add_security_group(client_security_group)
+
+        # Commands to install dependencies
+        instance.user_data.add_commands(
+            "yum install java-1.8.0 -y",
+            f'wget https://archive.apache.org/dist/kafka/{constants["KAFKA_VERSION"]}/{constants["KAFKA_DOWNLOAD_VERSION"]}.tgz',
+            f"tar -xzf {constants['KAFKA_DOWNLOAD_VERSION']}.tgz",
+            f"./{constants['KAFKA_DOWNLOAD_VERSION']}/bin/kafka-topics.sh --create --zookeeper {zookeeper} --replication-factor 2 --partitions 1 --topic {constants['MSK_TOPIC']}",
+        )
+
+
+class MskBroker(NestedStack):
+    def __init__(self, scope: Construct, construct_id: str, vpc, client_subnet, **kwargs):
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Create the MSK cluster with SASL/SCRAM authentication
+        self.cluster = msk.Cluster(self, "Cluster",
+            cluster_name="iotCluster",
+            kafka_version=msk.KafkaVersion.V2_8_1,
+            vpc=vpc,
+            encryption_in_transit=msk.EncryptionInTransitConfig(
+                client_broker=msk.ClientBrokerEncryption.TLS
+            ),
+            client_authentication=msk.ClientAuthentication.sasl(
+                scram=True
+            ),    
+        )
+
+        # Enable MSK cluster connection  
+        self.cluster.connections.allow_from(
+            ec2.Peer.ipv4("0.0.0.0/0"),
+            ec2.Port.tcp(2181))
+        self.cluster.connections.allow_from(
+            ec2.Peer.ipv4("0.0.0.0/0"),
+            ec2.Port.tcp(9096))
+        
+        # EC2 Instance in the public subnet used to create the topics
+        client = MSKClient(self, "MskClient", 
+            vpc=vpc, 
+            client_subnet=client_subnet, 
+            zookeeper=self.cluster.zookeeper_connection_string)
+
+
+class MskDemoStack(Stack):
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # 1 Public Subnet and 2 Private
```

Comment:
Explain what these subnets are used for so that customers understand the placement of resources within them.